packages/react-scripts/template/src/components/AppHeader.jsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
```

Comment:
I broke out this logic (and the logic for the logo) into separate components, only to help show a better separation of concerns. this might be overdoing it, I dunno.

Reply:
👍  I like it

Reply:
The AppHeader may be a bit overkill, but I think it demonstrates writing a wrapper component that just applies styles. If that's our pattern for doing that, then I think it's a good separation and demo usage. Their demo page is a bit unusual in that the header is the entire page, so it's a bit of a misnomer, in real life the header would just be some piece at the top and some content below and the abstraction would be more useful.
I personally feel our starter page should have more typical content in it and be more demonstrative of good structure, but we'll need to discuss that. Not relevant for this PR.

Reply:
We should probably author our own demo landing page, one without the spinning logo. Maybe one with some actual routing going on? Maybe something similar to the old frontier boilerplate landing page? Maybe something simpler?

Reply:
Will address in a separate PR